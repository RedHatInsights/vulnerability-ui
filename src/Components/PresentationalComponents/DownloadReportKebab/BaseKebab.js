/* eslint-disable camelcase */
import { Dropdown, KebabToggle } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React, { useState } from 'react';

const BaseKebab = ({ dropdownItems, disabled, ...props }) => {
    const [isOpen, setOpen] = useState(false);

    function handleOnKebabToggle(kebabState) {
        setOpen(kebabState);
    }

    function handleOnKebabSelect() {
        setOpen(!isOpen);
    }

    return (
        <div>
            <Dropdown
                onSelect={handleOnKebabSelect}
                toggle={<KebabToggle onToggle={handleOnKebabToggle} isDisabled={disabled} />}
                isOpen={isOpen}
                isPlain
                dropdownItems={dropdownItems}
                {...props}
            />
        </div>
    );
};

BaseKebab.propTypes = {
    dropdownItems: propTypes.array,
    disabled: propTypes.bool
};

export default BaseKebab;
