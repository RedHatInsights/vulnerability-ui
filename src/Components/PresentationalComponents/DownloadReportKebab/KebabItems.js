/* eslint-disable camelcase */
import { DropdownItem } from '@patternfly/react-core';
import React from 'react';

export const kebabItemDownloadJSON = (downloadReport, { ...props }) => (
    <DropdownItem key="json" component="button" onClick={() => downloadReport('json')} {...props}>
        Export as JSON
    </DropdownItem>
);

export const kebabItemDownloadCSV = (downloadReport, { ...props }) => (
    <DropdownItem key="csv" component="button" onClick={() => downloadReport('csv')} {...props}>
        Export as CSV
    </DropdownItem>
);

export const kebabItemEditBusinessRisk = (showBusinessRiskModal, cves, { ...props }) => (
    <DropdownItem key="editBusinessRisk" component="button" onClick={() => showBusinessRiskModal(cves)} {...props}>
        Edit business risk
    </DropdownItem>
);

export const kebabItemEditStatus = (showStatusModal, cves, inventoryIds,  { ...props }) => (
    <DropdownItem key="editStatus" component="button" onClick={() => showStatusModal(cves, inventoryIds)} {...props}>
        Edit status
    </DropdownItem>
);

export const kebabItemExcludeSystemAnalysis = (doOptOut, isOptOut,  { ...props }) => (
    <DropdownItem key="do_opt_out" component="button" onClick={() => doOptOut()} {...props}>
        {isOptOut ? 'Resume analysis for system' : 'Exclude system from analysis'}
    </DropdownItem>
);

export const kebabItemExcludeSystem = (doOptOut, isOptOut,  { ...props }) => (
    <DropdownItem key="do_opt_out" component="button" onClick={() => doOptOut()} {...props}>
        {isOptOut ? 'Resume analysis for system' : 'Exclude system from analysis'}
    </DropdownItem>
);

export const kebabItemToggleExcludeSystemDisplay = (handleOptOut, isOptedOut,  { ...props }) => (
    <DropdownItem key="toggle_opt_out" component="button" onClick={() => handleOptOut()} {...props}>
        {isOptedOut ? 'Hide systems excluded from analysis' : 'Show systems excluded from analysis'}
    </DropdownItem>
);
