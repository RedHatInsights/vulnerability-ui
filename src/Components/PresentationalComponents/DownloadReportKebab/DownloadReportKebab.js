import propTypes from 'prop-types';
import React from 'react';
import BaseKebab from './BaseKebab';
import { kebabItemDownloadCSV, kebabItemDownloadJSON } from './KebabItems';

const DownloadReportKebab = ({ downloadReport }) => {
    const dropdownItems = [kebabItemDownloadJSON(downloadReport), kebabItemDownloadCSV(downloadReport)];
    return (
        <div>
            <BaseKebab dropdownItems={dropdownItems} />
        </div>
    );
};

DownloadReportKebab.propTypes = {
    downloadReport: propTypes.func
};

export default DownloadReportKebab;
