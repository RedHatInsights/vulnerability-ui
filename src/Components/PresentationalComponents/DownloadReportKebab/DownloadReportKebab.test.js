import React from 'react';
import DownloadReportKebab from './DownloadReportKebab';
import toJson from 'enzyme-to-json';

describe('DownloadReportKebab component', () => {
    it('Should render without params', () => {
        const component = mount(<DownloadReportKebab/>);
        expect(toJson(component)).toMatchSnapshot();
    });
});
