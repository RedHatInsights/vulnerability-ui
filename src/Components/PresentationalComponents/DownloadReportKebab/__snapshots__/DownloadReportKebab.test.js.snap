// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadReportKebab component Should render without params 1`] = `
<DownloadReportKebab>
  <div>
    <BaseKebab
      dropdownItems={
        Array [
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Export as JSON"
              description="Kebab item"
              id="kebab.exportAsJson"
            />
          </DropdownItem>,
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Export as CSV"
              description="Kebab item"
              id="kebab.exportAsCsv"
            />
          </DropdownItem>,
        ]
      }
    >
      <div>
        <Dropdown
          dropdownItems={
            Array [
              <DropdownItem
                component="button"
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Export as JSON"
                  description="Kebab item"
                  id="kebab.exportAsJson"
                />
              </DropdownItem>,
              <DropdownItem
                component="button"
                onClick={[Function]}
              >
                <FormattedMessage
                  defaultMessage="Export as CSV"
                  description="Kebab item"
                  id="kebab.exportAsCsv"
                />
              </DropdownItem>,
            ]
          }
          isOpen={false}
          isPlain={true}
          onSelect={[Function]}
          toggle={
            <KebabToggle
              onToggle={[Function]}
            />
          }
        >
          <DropdownWithContext
            autoFocus={true}
            className=""
            direction="down"
            dropdownItems={
              Array [
                <DropdownItem
                  component="button"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="Export as JSON"
                    description="Kebab item"
                    id="kebab.exportAsJson"
                  />
                </DropdownItem>,
                <DropdownItem
                  component="button"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="Export as CSV"
                    description="Kebab item"
                    id="kebab.exportAsCsv"
                  />
                </DropdownItem>,
              ]
            }
            isGrouped={false}
            isOpen={false}
            isPlain={true}
            menuAppendTo="inline"
            onSelect={[Function]}
            position="left"
            toggle={
              <KebabToggle
                onToggle={[Function]}
              />
            }
          >
            <div
              className="pf-c-dropdown"
              data-ouia-component-id="OUIA-Generated-Dropdown-1"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <KebabToggle
                aria-haspopup={true}
                getMenuRef={[Function]}
                id="pf-dropdown-toggle-id-1"
                isOpen={false}
                isPlain={true}
                key=".0"
                onEnter={[Function]}
                onToggle={[Function]}
                parentRef={
                  Object {
                    "current": <div
                      class="pf-c-dropdown"
                      data-ouia-component-id="OUIA-Generated-Dropdown-1"
                      data-ouia-component-type="PF4/Dropdown"
                      data-ouia-safe="true"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Actions"
                        class="pf-c-dropdown__toggle pf-m-plain"
                        id="pf-dropdown-toggle-id-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 192 512"
                          width="1em"
                        >
                          <path
                            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                          />
                        </svg>
                      </button>
                    </div>,
                  }
                }
              >
                <Toggle
                  aria-haspopup={true}
                  aria-label="Actions"
                  bubbleEvent={false}
                  className=""
                  getMenuRef={[Function]}
                  id="pf-dropdown-toggle-id-1"
                  isActive={false}
                  isDisabled={false}
                  isOpen={false}
                  isPlain={true}
                  isPrimary={false}
                  isSplitButton={false}
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    Object {
                      "current": <div
                        class="pf-c-dropdown"
                        data-ouia-component-id="OUIA-Generated-Dropdown-1"
                        data-ouia-component-type="PF4/Dropdown"
                        data-ouia-safe="true"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Actions"
                          class="pf-c-dropdown__toggle pf-m-plain"
                          id="pf-dropdown-toggle-id-1"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 192 512"
                            width="1em"
                          >
                            <path
                              d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                            />
                          </svg>
                        </button>
                      </div>,
                    }
                  }
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label="Actions"
                    className="pf-c-dropdown__toggle pf-m-plain"
                    disabled={false}
                    id="pf-dropdown-toggle-id-1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    <EllipsisVIcon
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                    >
                      <svg
                        aria-hidden={true}
                        aria-labelledby={null}
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style={
                          Object {
                            "verticalAlign": "-0.125em",
                          }
                        }
                        viewBox="0 0 192 512"
                        width="1em"
                      >
                        <path
                          d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                        />
                      </svg>
                    </EllipsisVIcon>
                  </button>
                </Toggle>
              </KebabToggle>
            </div>
          </DropdownWithContext>
        </Dropdown>
      </div>
    </BaseKebab>
  </div>
</DownloadReportKebab>
`;
